pkg/operator/ceph/cluster/watcher.go

```diff
@@ -65,12 +74,18 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
 	if !ok {
 		return false
 	}
+
+	// Get CephCluster
+	cluster := c.getCephCluster()
+
+	if err := c.handleNodeFailure(ctx, cluster, node); err != nil {
+		logger.Errorf("failed to handle node failure. %v", err)
```

Comment:
How about adding a comment here about how we intentionally continue instead of failing the node reconcile.